### ISSUE TITLE
Error plotting results from mif2()

### ISSUE THREAD

**Issue by dvassilev2** [kingaa/pomp#3]
Hi,
I'm getting an error when trying to plot diagnostic charts after running mif2(). It's quite possible that it's me doing something wrong but it's not at all obvious from the error. So if it turns out not to be a bug, it would be nice to get an error message that allows me to fix the issue myself. 

mif2() runs without any warnings and the estimated parameter values make sense.

``` R
...

mif1 <- foreach(i = 1:10, .combine = c) %dopar% {
  mif2(model, Nmif = 100, start = perturbCoefs(coefs.true, sdCoefs), transform=TRUE,
       Np = 2000, rw.sd=sdCoefs_rw, cooling.fraction.50 = 0.05, verbose=TRUE)
}

pf1 <- foreach(mf = mif1, .combine = c) %dopar% {
  pf <- replicate(n = 10, logLik(pfilter(mf, Np = 10000)))
  logmeanexp(pf)
}

plot(mif1)
Hit <Return> to see next plot: 
Error in object@filter.mean[pars, ] : subscript out of bounds
```

The effective sample size chart is displayed but nothing else. Let me know if you'd like the complete data + model code so that you can replicate the problem.

Thanks

**Comment by kingaa** [comment]
Yes, would you please send a replicable example.  I will see, too, what can be done about making the error message more informative.

**Comment by dvassilev2** [comment]
I emailed the code + data to your umich.edu address.

**Comment by kingaa** [comment]
This is a bug.  In the case of a scalar state variable, the 'filter.mean' method was dropping the singleton dimension of the filter.mean array, in conflict with the assumption of the plotting method.  This has been fixed in version 1.2.1.2, which I'll commit to github now.  If you install this version and get another error, please let me know!

Thanks for your very efficient bug report!

**Comment by dvassilev2** [comment]
You are welcome and thank you for the quick fix!
